### DISCUSSION TITLE
FNIRSI GC-01: Data communications

### DISCUSSION THREAD

**Discussion by ioderey** [Gissio/radpro#105] [Q&A] [answered]
Hello, Gissoi, your work has taken GC-01 to a new, high level! I may be a noob, but I have not been able to establish a connection between the device and the computer in order to use the opportunity to change some settings of the device while it is running. Connecting a working device with the same cable that I successfully update the firmware, I receive a windows 10 message that the device ID has not been determined, respectively, I see an unknown usb device in the device manager. Question: do I need any additional devices besides the USB cable connected to the USb connector of the device to change its settings on the go, or do I need to connect to the pins on the board via ST-link? My chip is CH32f103.
I would like to see a simple guide for working with the device via a USB cable: which SW and HW are needed for this?

**Comment by Gissio** [comment, marked as answer]
Please make sure you read the manual: https://github.com/Gissio/radpro/blob/main/docs/devices/FNIRSI%20GC-01/install.md#fnirsi-gc-01-with-a-ch32f103r8-wch-microprocessor

**Reply by ioderey** [reply]
I'm talking about it, from the manual: 
"Communication protocol
You can use either a serial port or a SWD connection to establish a connection with the Rad Pro device." 
Which of the HW and SW should I use to send from PC to a device like "GET DeviceID\r\n"?

**Reply by ioderey** [reply]
>>you must enable “Data mode” in the settings to establish a USB data connection.
data mode is on, but windows say " device ID has not been determined" . 
Perhaps I need some driver?

**Reply by Gissio** [reply]
With the FNIRSI GC-01 you just need a USB data cable.

With a CH32 microcontroller, you also need to enable "Data mode" in Rad Pro's settings. Once you are done, turn off "Data mode" in order to save power.

**Reply by ioderey** [reply]
Hi, Gissio! I update radpro to v2.01, USB connection present now, all right.
